src/ekka_autoheal.erl

```diff
@@ -122,20 +139,21 @@ compare_view({Running1, _} , {Running2, _}) ->
 coordinator([{Nodes, _} | _]) ->
     ekka_membership:coordinator(Nodes).
 
--spec heal_partition(list()) -> list(node()).
-heal_partition([]) ->
-    [];
-%% All nodes connected.
-heal_partition([{_, []}]) ->
-    [];
-%% Partial partitions happened.
-heal_partition([{Nodes, []}|_]) ->
+heal_partition([{Nodes, []} | _] = SplitView) ->
+    %% Symmetric partition.
+    ?LOG(info, "Healing partition: ~p", [SplitView]),
     reboot_minority(Nodes -- [node()]);
-heal_partition([{Majority, Minority}, {Minority, Majority}]) ->
+heal_partition([{Majority, Minority}, {Minority, Majority}] = SplitView) ->
+    %% Symmetric partition.
+    ?LOG(info, "Healing partition: ~p", [SplitView]),
     reboot_minority(Minority);
-heal_partition(SplitView) ->
-    ?LOG(critical, "Cannot heal the partitions: ~p", [SplitView]),
-    error({unknown_splitview, SplitView}).
+heal_partition([{_Nodes, Stopped} | _Asymmetric] = SplitView) ->
+    %% Asymmetric partitions.
+    %% Start with rebooting known stopped nodes. If this won't be enough, retry mechanism
+    %% in `ekka_node_monitor:handle_info({mnesia_system_event, ...}` should then launch
+    %% new iteration.
+    ?LOG(info, "Trying to heal asymmetric partition: ~p", [SplitView]),
+    reboot_minority(Stopped).
```

Comment:
This might bring the cluster back, but the data might be corrupted.
Here is what happened: in a 4-node cluster, each node sees one node is out side of the cluster, e.g.

```
A: {[A, B, C], [D]}
B: {[B, C, D], [A]}
C: {[A, B, D], [B]}
D: {[A, B, D], [C]}
```

After node `B` was restarted, the cluster immediately healed.
But clients connected to node `B` cannot receive publishes from other nodes anymore.
The cluster was recovered 6 hours later with a rolling restart.
Unfortunately, no inspection was made before the rolling restart, so we don't know what exactly went wrong, but likely it's the routing information corrupted for node B in other nodes.